contracts/core/EntryPointSimulations.sol

```diff
@@ -0,0 +1,168 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.12;
+
+/* solhint-disable avoid-low-level-calls */
+/* solhint-disable no-inline-assembly */
+
+import "./EntryPoint.sol";
+import "../interfaces/IEntryPointSimulations.sol";
+
+/*
+ * This contract inherits the EntryPoint and extends it with the view-only methods that are executed by
+ * the bundler in order to check UserOperation validity and estimate its gas consumption.
+ */
+contract EntryPointSimulations is EntryPoint, IEntryPointSimulations {
```

Comment:
as per @yoavw 's suggestion, 
please add reverting constructor, with explanation that this code should only be used with "stateOverride"